src/app/auth/auth.service.ts

```diff
@@ -86,62 +87,39 @@ export class AuthService {
-    }).then(loggedin => {
-      return this.httpService.get<NameResponse>(
-        this.globalService.serverURL + '/displaynames/:users?users=' + username,
-        this.globalService.options).toPromise().then(names => {
-          if (names.displaynames.length > 0) {
-            user.displayname = names.displaynames[0].displayname;
-          }
-          return this.httpService.get<EmailResponse>(
-            this.globalService.serverURL + '/email/:user?user=' + username,
-            this.globalService.options).toPromise().then(email => {
-              user.email = email.email;
-              return this.httpService.get<PhoneResponse>(
-                this.globalService.serverURL + '/phone/:user?user=' + username,
-                this.globalService.options).toPromise().then(phone => {
-                  user.phone = phone.phone;
-                  return this.httpService.get<DobResponse>(
-                    this.globalService.serverURL + '/dob/:user?user=' + username,
-                    this.globalService.options).toPromise().then(dob => {
-                      user.birthday = dob.dob;
-                      return this.httpService.get<ZipCodeResponse>(
-                        this.globalService.serverURL + '/zipcode/:user?user=' + username,
-                        this.globalService.options).toPromise().then(zipcode => {
-                          user.zipcode = zipcode.zipcode;
-                          return this.httpService.get<StatusResponse>(
-                            this.globalService.serverURL + '/headlines/:users?users=' + username,
-                            this.globalService.options).toPromise().then(headlines => {
-                              if (headlines.headlines.length > 0) {
-                                user.status = headlines.headlines[0].headline;
-                              }
-                              return this.httpService.get<AvatarResponse>(
-                                this.globalService.serverURL + '/avatars/:users?users=' + username,
-                                this.globalService.options).toPromise().then(avatars => {
-                                  if (avatars.avatars.length > 0) {
-                                    user.avatar = avatars.avatars[0].avatar;
-                                  }
-                                }).then(() => {
-                                  if (loggedin) { this.makeNewUser(user); }
-                                  return loggedin;
-                                });
-                            });
-                        });
-                    });
-                });
-            });
-        });
+    return request.toPromise().then(res => {
+      console.log(res.headers.get('Token'));
+      if(res.body['result'] === true){
+        sessionStorage.setItem('Token',res.headers.get('Token'));
+        return true;
+      }else{
+        return false;
+      }
+    }).then(() => {return true;
```

Comment:
We need some conditions on whether returning true. See branch ```migrate``` for an updated version of this function.